### PR TITLE
[doc] Update the User Guide to refer to Meson commands.

### PR DESCRIPTION
**Pull request by mcy**
If I missed any guides, please point them out to me.

This change is pending #1040, which makes ./meson_init.sh idempotent (which makes writing guides involving meson_init.sh way easier).

### CODE REVIEW

File: doc/ug/getting_started_verilator.md [hunk 1]
```diff
@@ -23,16 +25,13 @@ There are 3 memory types: ROM, RAM and Flash.
 By default, the system will first execute out of ROM and then jump to flash.
 A program needs to be built for each until ROM functionality for code download is ready.
 
-For that purpose compile the demo program with "simulation" settings, which adjusts the frequencies to better match the simulation speed.
-In the instructions below, `SW_DIR` is a requirement argument, while `SW_BUILD_DIR` is not a required argument.
-If `SW_BUILD_DIR` argument is not supplied, the default location of the of output files are in `SW_DIR`
-Please see [SW build flow]({{< relref "sw/device/doc/sw_build_flow" >}}) for more details.
+For that purpose compile the demo program with "simulation" settings, which adjusts the frequencies to better match the simulation speed. 
+For more information on building software targets, [click here]({{< relref "getting_started_sw.md" >}}).
```
Comment (imphil) [inline doc/ug/getting_started_verilator.md]: nit: avoid `here` links; 
```suggestion
For more information on building software targets refer to the [Software Getting Started Guide]({{< relref "getting_started_sw.md" >}}).
```

File: sw/host/spiflash/README.md [hunk 2]
```diff
@@ -59,8 +60,8 @@ After the transmission is complete, you should be able to see the hello_world ou
 ```console
 $ cd ${REPO_TOP}
 $ build-verilator/sw/host/spiflash/spiflash \
```
Comment (imphil) [inline sw/host/spiflash/README.md]: Use the spiflash binary from build-bin here.

File: sw/host/spiflash/README.md [hunk 3]
```diff
@@ -73,5 +74,5 @@ If there are two FPGAs or multiple valid targets attached at the same time, it i
 ```console
 $ cd ${REPO_TOP}
 $ build-fpga/sw/host/spiflash/spiflash \
```
Comment (imphil) [inline sw/host/spiflash/README.md]: also use the one from build-bin here.

**Comment by tjaychen** [comment]
there's also https://github.com/lowRISC/opentitan/blob/master/doc/rm/ref_manual_fpga.md
but it is debateable if we should get rid of that doc altogether.

**Comment by mcy** [comment]
@tjaychen Added.

**Comment by imphil** [comment]
Thanks!

**Comment by asb** [comment]
As noted in #1154, I think this is incomplete as it leaves some software-related make invocations in the fpga getting started instructions.

**Pull request merged**